### DISCUSSION TITLE
No Composer update PRs

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#11003] [Request Help] [answered]
I've set up [Renovate on GitLab](https://gitlab.com/renovate-bot/renovate-runner) and it works very well for one repository.

However, for another repository there are no update PRs for Composer packages at all even though `composer outdated` clearly shows a long list of possible updates.

I've even [nuked the config and triggered another onboarding](https://docs.renovatebot.com/reconfigure-renovate/#nuke-config-and-re-onboard) but this also tells me this:

> It looks like your repository dependencies are already up-to-date and no Merge Requests will be necessary right away.

Here's my config:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":preserveSemverRanges",
    ":separatePatchReleases"
  ],
  "commitMessagePrefix": "[TASK] ",
  "commitMessageTopic": "{{depName}}",
  "commitMessageExtra": " ({{{currentValue}}} => {{{newValue}}})",
  "packageRules": [
    {
      "matchUpdateTypes": ["patch"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true,
      "gitLabAutomerge": true
    }
  ]
}
```

Even the following does not work and only suggests to pin all dependencies (default behavior of Renovate):

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ]
}
```

Here's an example for `symfony/console` from the `renovate-log.ndjson` which is currently installed at [5.3.2](https://packagist.org/packages/symfony/console#v5.3.2) but [5.3.4](https://packagist.org/packages/symfony/console#v5.3.4) is the latest:

```json
                    {
                        "depType":"require",
                        "depName":"symfony/console",
                        "currentValue":"^5.2",
                        "datasource":"packagist",
                        "lockedVersion":"5.3.2",
                        "depIndex":18,
                        "updates":[
                            
                        ],
                        "warnings":[
                            
                        ],
                        "versioning":"composer",
                        "sourceUrl":"https://github.com/symfony/console",
                        "homepage":"https://symfony.com",
                        "currentVersion":"5.3.2",
                        "fixedVersion":"5.3.2"
                    },
```

How can there be no updates here? Even a `composer up --dry-run` works just fine.

**Comment by viceice** [comment, marked as answer]
The new version matches your range, so renovate won't update it by default.

Checkout range strategy and lockfile maintenance.

**Reply by rarkins** [reply]
`rangeStrategy` in the docs

**Reply by mbrodala** [reply]
Thanks, that was the issue!

After dropping the `":preserveSemverRanges"` (`"rangeStrategy": "replace"`) preset and adding `"rangeStrategy": "update-lockfile"` instead the onboarding PR now correctly suggests pending Composer updates. :-)

In the other project which worked fine I already had `"rangeStrategy": "update-lockfile"`.

Maybe the `:preserveSemverRanges` preset name is a bit misleading since, at least for me, preserving semver ranges should allow patch updates.

**Reply by rarkins** [reply]
Yes, we'll probably update the terminology of `rangeStrategy` in future (and maybe its name too). rangeStrategy=replace essentially means "no PR unless the package file needs updating".
